### PR TITLE
using multipleItemViews

### PR DESCRIPTION
**Pull request by andych008**

### CODE REVIEW

File: app/src/main/java/me/tatarka/bindingcollectionadapter/sample/FragmentListView.java [hunk 1]
```diff
@@ -20,7 +22,14 @@
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setRetainInstance(true);
-        viewModel = new ViewModel(true);
+
+        ObservableList<ItemViewModel> items = new ObservableArrayList<>();
+        items.add(new HeaderItemViewModel("我是header"));
```
Comment (Dev-iL) [inline app/src/main/java/me/tatarka/bindingcollectionadapter/sample/FragmentListView.java]: Hard-coded strings are not a good idea.... If you must put a string somewhere, you should use the `strings.xml` resource file (which would also allow localization; unfortunately not everybody reads Hanzi...)

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Wed Apr 10 15:27:10 PDT 2013
+#Mon Dec 28 10:00:20 PST 2015
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-all.zip
```
Comment (Dev-iL) [inline gradle/wrapper/gradle-wrapper.properties]: There's already Gradle 2.12, why not use _that_?

File: app/src/main/java/me/tatarka/bindingcollectionadapter/sample/FragmentListView.java (hunk 1 above)
Comment (andych008) [inline app/src/main/java/me/tatarka/bindingcollectionadapter/sample/FragmentListView.java]: It's my fault.I'll fix it.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (andych008) [inline gradle/wrapper/gradle-wrapper.properties]: Android Studio 2.0's default gradle version is 2.10. so I think most of us will use 2.10.

**Comment by andych008** [comment]
I also have a question. for binding multipleItemViews, Do you think ObservableList<ItemViewModel> or    binding  ObservableList<Object>, whick one is a good idea?

File: app/proguard-rules.pro [hunk 3]
```diff
@@ -10,7 +10,7 @@
 # Add any project specific keep options here:
 
 # If your project uses WebView with JS, uncomment the following
-# and specify the fully qualified class name to the JavaScript interface
+# and specify the fully qualified class text to the JavaScript interface
```
Comment (evant) [inline app/proguard-rules.pro]: accidental refactor-rename?

File: app/src/main/java/me/tatarka/bindingcollectionadapter/sample/NormalItemViewModel.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Created by evan on 6/14/15.
+ */
+public class NormalItemViewModel extends ItemViewModel {
+    public final boolean checkable;
+    private boolean checked;
+
+    public NormalItemViewModel(int index, boolean checkable) {
+        super(index+" a item");
```
Comment (evant) [inline app/src/main/java/me/tatarka/bindingcollectionadapter/sample/NormalItemViewModel.java]: Should be `"Item " + index`, but why did you remove this from the layout? The header already has a separate layout so it can compute it differently.

File: app/src/main/java/me/tatarka/bindingcollectionadapter/sample/ViewModel.java [hunk 5]
```diff
@@ -13,32 +13,37 @@
...
+        this.checkable = checkable;
+        this.items = items;
+
+    }
 
     /**
      * ItemView of a single type
      */
-    public final ItemView singleItemView = ItemView.of(BR.item, R.layout.item);
+    public final ItemView singleItemView = ItemView.of(me.tatarka.bindingcollectionadapter.sample.BR.item, R.layout.item);
```
Comment (evant) [inline app/src/main/java/me/tatarka/bindingcollectionadapter/sample/ViewModel.java]: No reason to add the whole package name here.

File: app/src/main/java/me/tatarka/bindingcollectionadapter/sample/ViewModel.java [hunk 6]
```diff
@@ -65,12 +70,12 @@ public long getItemId(int position, ItemViewModel item) {
     public final BindingViewPagerAdapter.PageTitles<ItemViewModel> pageTitles = new BindingViewPagerAdapter.PageTitles<ItemViewModel>() {
         @Override
         public CharSequence getPageTitle(int position, ItemViewModel item) {
-            return "Item " + (item.getIndex() + 1);
+            return "Item " + (item.getName() + 1);
```
Comment (evant) [inline app/src/main/java/me/tatarka/bindingcollectionadapter/sample/ViewModel.java]: With your changes above won't this say "Item 2 a item"?

File: app/src/main/res/values-zh/strings.xml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="app_name">BindingCollectionAdapter</string>
+    <string name="header">我是header</string>
```
Comment (evant) [inline app/src/main/res/values-zh/strings.xml]: I understand wanting to have this, but it think everything should be translated or none at all, and I don't think I'd be able to maintain this translation.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (evant) [inline gradle/wrapper/gradle-wrapper.properties]: Yeah I'm fine with this

**Comment by evant** [comment]
Sorry for taking so long to take a look. In addition to my above comments can you squash your commits? I think this is on the right track overall.

**Comment by andych008** [comment]
thnks for your talking. next time my codes will be improved.
